heat/core/manipulations.py

```diff
@@ -1824,10 +1848,18 @@ def reshape_argsort_counts_displs(
     # Reshape local tensor
     data = data.reshape(local_shape)
 
-    return factories.array(data, dtype=a.dtype, is_split=new_split, device=a.device, comm=a.comm)
+    return DNDarray(
+        data,
+        tuple(shape),
+        dtype=a.dtype,
+        split=new_split,
+        device=a.device,
+        comm=a.comm,
+        balanced=True,
+    )
 
 
-DNDarray.reshape = lambda self, shape, axis=None: reshape(self, shape, axis)
+DNDarray.reshape = lambda self, *shape, **kwargs: reshape(self, *shape, **kwargs)
```

Comment:
This works but passing shape elements as single arguments is so much slower than passing a tuple. Need help here @Markus-Goetz @coquelin77